[Donovan Lee] iP

24Donovan24 opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :stuck_out_tongue_winking_eye:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

24Donovan24 committed:
Implemented Level-1

24Donovan24 committed:
Implemented Level-2

24Donovan24 committed:
Implemented Level-3 and minor changes to conditionals and scanner usage

24Donovan24 committed:
Changed if else conditionals to switch statements

24Donovan24 committed:
Implemented subclasses of Task

24Donovan24 committed:
Implemented Level-4

24Donovan24 committed:
Edited code for testing

24Donovan24 committed:
Implemented A-TextUiTesting

24Donovan24 committed:
Created separate class files for each class

24Donovan24 committed:
Created DukeException class to handle exceptions specific to Duke

24Donovan24 committed:
Implemented Level-5

24Donovan24 committed:
Implemented Level-6

24Donovan24 committed:
Changed Task to an abstract class

24Donovan24 committed:
Added javadocs to methods

24Donovan24 committed:
Abstracted various methods

24Donovan24 committed:
Handled exception when task id is an inavlid number

24Donovan24 committed:
Handled exception when deleting task with invalid number

24Donovan24 committed:
Removed unnecessary code

24Donovan24 committed:
Abstracted bots actions into a run method

24Donovan24 committed:
Removed unnecessary code

24Donovan24 committed:
Modified EXPECTED.TXT

24Donovan24 committed:
Add LINE_BREAK constant and minor style fix

24Donovan24 committed:
Fix minor coding style

24Donovan24 committed:
Add savedString function to Task and its subclasses

24Donovan24 committed:
Implement Level-7

24Donovan24 committed:
Implement Level-8

24Donovan24 committed:
Merge branch 'branch-Level-8'

24Donovan24 committed:
Implement A-MoreOOP

24Donovan24 committed:
Implement A-Packages

24Donovan24 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

24Donovan24 committed:
Fix minor bug

24Donovan24 committed:
Implement A-JUnit

24Donovan24 committed:
Fix some OOP

24Donovan24 committed:
Fix some bug

24Donovan24 committed:
Add JavaDoc for all non-private classes/methods

24Donovan24 committed:
Update some coding standard

24Donovan24 committed:
Implement Level-9

24Donovan24 committed:
Merge with branch-A-CodingStandard

24Donovan24 committed:
Merge with branch-Level-9

24Donovan24 committed:
Fix minor bug

24Donovan24 committed:
Add more JavaDocs

24Donovan24 committed:
Add more OOP and handle more exceptions

24Donovan24 committed:
Fix minor bug

24Donovan24 committed:
Fix some JavaDocs

ElijahS67 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,132 @@
+    public void printTodo(Task todo, int size) {
+        System.out.printf(LINE_BREAK
+                + "\n\tGot it. I've added this task:\n\t%s\n\tNow you have %d tasks in the list.\n"
+                        + LINE_BREAK + "\n", todo, size);
```
Nice use of line breaks to improve readability!

ElijahS67 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,59 @@
+     *
+     * @return String to be displayed in the file.
+     */
+    public abstract String savedString();
```
Perhaps toSavedString() is a better method name?

24Donovan24 replied:
Thanks for the review! I will implement the following suggestion!

ElijahS67 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,97 @@
+            try {
+                switch (text) {
+                //Handle case when task aTodo
+                case "todo" :
```
Perhaps there should not be a spacing between "todo" and the colon?

24Donovan24 replied:
Thanks for the review! I will make that change!

ElijahS67 reviewed (commented):
Looks good! Almost LGTM!

24Donovan24 committed:
Implement Level 10

24Donovan24 committed:
Add JavaDocs for edited functions

24Donovan24 committed:
Fix some coding standards

24Donovan24 committed:
Fix some coding standard

24Donovan24 committed:
Fix some minor bug

24Donovan24 committed:
Add Checkstyle

24Donovan24 committed:
Fix checkstyle violations

24Donovan24 committed:
Add assertions

24Donovan24 committed:
Merge pull request #1 from 24Donovan24/branch-A-Assertions

Add assertions

24Donovan24 committed:
Fix code quality

24Donovan24 committed:
Merge pull request #2 from 24Donovan24/branch-A-CodeQuality

Fix code quality

24Donovan24 committed:
Add extension C-Statistics

24Donovan24 committed:
Merge pull request #3 from 24Donovan24/C-Statistics

Add extension C-Statistics

24Donovan24 committed:
Fix missing user input bug

Some commands like mark, unmark and delete require users to specify the
index of the task. Currently, if a user omit the index, there will be no
output.

A message should be sent to remind users to retype their command with an
appropriate index.

To fix the bug of a missing index, a
checkForMissingArgs() method is introduced to remind user to key in an
index for those commands should they left it out.

24Donovan24 committed:
Improve GUI

24Donovan24 committed:
Edit personality of bot

24Donovan24 committed:
Handle more error

24Donovan24 committed:
Create User Guide
